### ISSUE TITLE
Create Two-Tower Model

### ISSUE THREAD

**Issue by AIHenry** [ThirdAILabs/Universe#170]
- [x] Decide on the actual network model for Two-Tower. For instance, we could:
    - define a new "paired" layer
    - define a new network that connects two Fully Connected BOLT networks
    - add a method to FullyConnectedLayer that computes error with another instance of FullyConnectedLayer 
- [ ] Decide on the loss function/metrics for Two-Tower. Tensorflow uses TopK retrieval accuracy as loss function.
- [ ] Get Movielens-100k dataset ready to feed into our model. 
- [x] Coding. Estimated timeline: Two weeks for a working version. 
- [ ] Testing. Estimated timeline: One week for a good accuracy on Movielens-100k.
